starters/cms/src/data.ts

```diff
@@ -38,14 +40,72 @@ export const dataSourceOptions = [
  *   ]
  * }
  */
+
+type ExtendedManagedCollection = ManagedCollection & {
+    dataSourceId: string | null
+}
+
+let collections: ManagedCollection[] = []
+let collectionsWithDataSourceId: ExtendedManagedCollection[] = []
+
+if (framer.mode === "syncManagedCollection" || framer.mode === "configureManagedCollection") {
+    collections = await framer.getManagedCollections()
+    collectionsWithDataSourceId = await Promise.all(
+        collections.map(async collection => {
+            const dataSourceId = await collection.getPluginData(PLUGIN_KEYS.DATA_SOURCE_ID)
+            return { ...collection, dataSourceId } as ExtendedManagedCollection
```

Comment:
This cast is dangerous. `dataSourceId` should be type checked

Reply:
I removed the typecasting